src/metabase/lib/breakout.cljc

```diff
@@ -75,24 +75,22 @@
 
   ([query        :- ::lib.schema/query
     stage-number :- :int]
-   (let [cols (let [stage   (lib.util/query-stage query stage-number)
-                    options {:include-implicitly-joinable-for-source-card? false}]
-                (lib.metadata.calculation/visible-columns query stage-number stage options))]
-     (when (seq cols)
-       (let [matching (into {} (keep-indexed (fn [index a-breakout]
-                                               (when-let [col (lib.equality/find-matching-column
-                                                               query stage-number a-breakout cols
-                                                               {:generous? true})]
-                                                 [col [index a-breakout]]))
-                                             (or (breakouts query stage-number) [])))]
-         (mapv #(let [[pos a-breakout] (matching %)
-                      binning (lib.binning/binning a-breakout)
-                      {:keys [unit]} (lib.temporal-bucket/temporal-bucket a-breakout)]
-                  (cond-> (assoc % :lib/hide-bin-bucket? true)
-                    binning (lib.binning/with-binning binning)
-                    unit (lib.temporal-bucket/with-temporal-bucket unit)
-                    pos (assoc :breakout-position pos)))
-               cols))))))
+   (let [columns (let [stage   (lib.util/query-stage query stage-number)
+                       options {:include-implicitly-joinable-for-source-card? false}]
+                   (lib.metadata.calculation/visible-columns query stage-number stage options))]
+     (when (seq columns)
+       (let [existing-breakouts         (breakouts query stage-number)
+             column->breakout-positions (group-by
+                                         (fn [position]
+                                           (lib.equality/find-matching-column query
+                                                                              stage-number
+                                                                              (get existing-breakouts position)
+                                                                              columns))
+                                         (range (count existing-breakouts)))]
+         (mapv #(let [positions  (column->breakout-positions %)]
+                 (cond-> (assoc % :lib/hide-bin-bucket? true)
```

Comment:
do not set `binning` and `unit` here because there is no 1:1 matching between breakouts and columns anymore